inject/codegen-jakarta/src/main/java/io/helidon/inject/codegen/jakarta/CdiTypes.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.inject.codegen.jakarta;
+
+import io.helidon.common.types.TypeName;
+
+final class CdiTypes {
```

Comment:
I understand why you did it - but it does seems wrong to have Cdi blend into the base processing / libraries.

If we think ahead there are other frameworks and types that we should be dealing with - so calling out Cdi as being special is setting a bad precedent for the longer term. I realize CDI is "special" to us now since Helidon and microprofile are sort of intertwined at the moment - but we know we want to eventually keep these things separate.

Also, we only really need `ApplicationScoped` in processing unless you plan to fail compilation if these other types are defined.  However, the later would have required you to setup a `Set` of unsupported type names to trigger the APT exception, and I did not see that here.  So given that we should either add the Set and detect these bad types, or we reduce it to just the types we need for pico.

Reply:
This is something that you do in existing `UnsupportedConstructsProcessor`, I just moved it to a different place.
I think we can remove the whole thing, including mapping of `@ApplicationScoped`, as that should not really be used (again, this is me keeping the existing behavior). Just say the word, and I will remove both the CDI types and the unsupported codegens from my PR.